python/kserve/kserve/storage/storage.py

```diff
@@ -278,8 +277,29 @@ def _download_s3(uri, temp_dir: str) -> str:
                 temp_dir = Storage._unpack_archive_file(target, mimetype, temp_dir)
         return temp_dir
 
+    @staticmethod
+    def _download_hf(uri, temp_dir: str) -> str:
+        from transformers import AutoTokenizer, AutoConfig, AutoModel
+
+        components = uri.split("://")[1].split("/")
+
+        repo = components[0]
+        model, _, hash_value = components[1].partition(":")
+        revision = hash_value if hash_value else None
+
+        tokenizer = AutoTokenizer.from_pretrained(f"{repo}/{model}", revision=revision)
+        model_config = AutoConfig.from_pretrained(f"{repo}/{model}", revision=revision)
+        model = AutoModel.from_config(model_config)
```

Comment:
custom code models must set the trust_remote_code argument to true.
need to use either the inference service's model arguments or the HF hub model API.